### DISCUSSION TITLE
Directus as website CMS

### DISCUSSION THREAD

**Discussion by kees-tce** [directus/directus#7097] [Archive]
Hi all!

Just trying to pick some brains. I'm thinking about using Directus as a headless cms for our website.

Any collection model examples out there? How do we store content best? MD? HTML? And with what metadata?

Furthermore, for the frontend I was thinking about a SSR Nuxt application, but perhaps there are some other/easier frontend frameworks for this?

Any input is more than welcome!

Kees.

**Comment by rijkvanzanten** [comment]
That's a very opinionated question!

My personal preference: save body copy in markdown so it's easier to convert later on for other uses. SSR Nuxt would definitely work, though if it's a simpler static site, you might want to look at a SSG instead of a server-side rendered approach. I believe Nuxt has a way to output as a static site, or you could look at something like [eleventy](https://11ty.io). We'll actually be setting up a dummy project shortly, so that could be a good time to cross reference our research on the topic 👍🏻

**Reply by philleepflorence** [reply]
Follow-up Question(s) - @kees-tce : 

Any example of use-cases or scenarios?
<details>
    <summary>Purpose or function of the website or user application...</summary>
    <p>...data or content storage, which could affect a whole lot of opinions, choices, and implementations.
Marketing Website, Blogging Website, Social Application, Community Application, Productivity Application, Learning Application, Knowledge Application, and the list goes on...

> Used Directus for all of the above and would be curious to see how others have implemented.</p>
</details>

@rijkvanzanten - ❤️ reference to [Nuxt.JS](https://nuxtjs.org) (PWA, SSR, SSG, and SPA Use Cases).

**Reply by kees-tce** [reply]
Well it's supposed to be a "standard" B2B company website. Think [BTC](https:://btc.nl) or [Incentro](https://incentro.com) 

Partners, cases/projects, contact and service descriptions. Where cases/projects can/will update regularly.

**Reply by rijkvanzanten** [reply]
Ik zou dan een static site generator eens proberen! Kan je hosten op een service zoals Netlify, dus geen gezeur met servers en onderhoud. In Directus kan je dan een webhook instellen om je website opnieuw te bundelen zodra er een wijziging is in de content.

_I'd try an SSG in that case! You can host the output on a service like Netlify, so you'll prevent the hassle of having to deal with servers and maintenance. You can setup a webhook in Directus to rebuild your website on content changes._

**Reply by kees-tce** [reply]
Thanks! And thanks for the Dutch version ;)

Currently running Directus in Google Cloud Run, the static site will be hosted on Firebase Hosting. All is being linted/tested/deployed using GitHub actions. The webhook idea is great!!

For static site I think I'm going to try [Nuxt Content](https://content.nuxtjs.org/)!

**Comment by keesvanbemmel** [comment]
And as for the second part: are there any suggestions/best practices/examples on how to model the content collection?

**Reply by rijkvanzanten** [reply]
That really depends on what sort of website you're building! For a simple blog style website, I'd imagine you'd at least have an `articles` collection, maybe a `website_settings` singleton (for things like header logo etc), a `pages` collection, maybe with a many-to-any for "sections" so you end up with a page builder?

**Reply by keesvanbemmel** [reply]
Thanks @rijkvanzanten!
